src/ansi-c/library/stdlib.c

```diff
@@ -115,6 +115,15 @@ inline void *malloc(__CPROVER_size_t malloc_size)
   // realistically, malloc may return NULL,
   // and __CPROVER_allocate doesn't, but no one cares
   __CPROVER_HIDE:;
+
+  __CPROVER_assert(
+    malloc_size < ((__CPROVER_size_t)1
+                   << (__CPROVER_pointer_width - __CPROVER_object_bits)),
+    "max allocation size exceeded");
+  __CPROVER_assume(
+    malloc_size < ((__CPROVER_size_t)1
+                   << (__CPROVER_pointer_width - __CPROVER_object_bits)));
```

Comment:
Given that object offsets are signed (see #5096), I think this should be `malloc_size <= ((__CPROVER_size_t)1 << (__CPROVER_pointer_width - __CPROVER_object_bits - 1))`. Also we can use `<=` as the maximum allocation size is one larger than the largest offset we can represent.

Reply:
Nice catch, thanks.